OpenRA.Mods.Common/Traits/BotModules/BaseBuilderBotModule.cs

```diff
@@ -231,21 +233,47 @@ void SetRallyPointsForNewProductionBuildings(IBot bot)
 				if (rp.Actor.Owner != player)
 					continue;
 
-				if (rp.Trait.Path.Count == 0 || !IsRallyPointValid(rp.Trait.Path[0], rp.Actor.Info.TraitInfoOrDefault<BuildingInfo>()))
```

Comment:
I think it would still make sense to bail early if we already set a rally point:
```
if (rp.Trait.Path.Count > 0)
	continue;
```

Reply:
A rally point might become invalid when a new building gets placed, so I am keeping it for now.